Doc/library/string.rst

```diff
@@ -227,7 +227,9 @@ See also the :ref:`formatspec` section.
 
 The *field_name* itself begins with an *arg_name* that is either a number or a
 keyword.  If it's a number, it refers to a positional argument, and if it's a keyword,
-it refers to a named keyword argument.  If the numerical arg_names in a format string
+it refers to a named keyword argument. A key is treated as a number if
+``str(key).isdecimal()`` would evaluate to true.
```

Comment:
```suggestion
it refers to a named keyword argument. An *arg_name* is treated as a number if
a call to :meth:`str.isdecimal` on the string would return true.
```
Use the same term as earlier in the paragraph, and link to the method so that users can look up what it does.